cpp/src/groupby/common/type_info.hpp

```diff
@@ -78,6 +77,15 @@ struct target_type<SourceType, SUM,
   using type = int64_t;
 };
 
+// Always use `double` for MEDIAN. It is because it involves 
+// division ((a+b)/2) for cases where the number of rows is even.  
+template <typename SourceType>
+struct target_type<SourceType, MEDIAN> { using type = double; };
```

Comment:
This should use what is defined for `QUANTILE` since `MEDIAN` is just a special case of a quantile.  
```suggestion
struct target_type<SourceType, MEDIAN> { using type = target_type_t<SourceType, QUANTILE>; };
```